src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -128,15 +139,17 @@ public async Task StartAsync(Uri url, TransferFormat transferFormat, Cancellatio
                 if (!string.IsNullOrEmpty(accessToken))
                 {
                     // We can't use request headers in the browser, so instead append the token as a query string in that case
-                    if (_isRunningInBrowser)
+                    if (OperatingSystem.IsBrowser())
                     {
                         var accessTokenEncoded = UrlEncoder.Default.Encode(accessToken);
                         accessTokenEncoded = "access_token=" + accessTokenEncoded;
                         resolvedUrl = Utils.AppendQueryString(resolvedUrl, accessTokenEncoded);
                     }
                     else
                     {
+#pragma warning disable CA1416 // Analyzer bug
```

Comment:
This feels given you have an explicit `OperatingSystem.IsBrowser()` check in the if part. What are the platform attributes applied to `Options` and `SetRequestHeader` symbols?

Reply:
Seems `[System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]`
https://github.com/dotnet/runtime/pull/41963/files#diff-0f06b592b2ee5b30d95ae38efae16a85R49